File: mobile-client/android/UseCase/Message/Thread/ListMessageThreadsUseCase.kt
package app.mobiling.client.client.usecase.message.thread

import app.mobiling.client.client.contract.message.thread.MessageThreadSummary
import app.mobiling.client.client.data.message.thread.MessageThreadGateway

/**
 * Marketing America Corp. Oleksandr Tishchenko
 */
class ListMessageThreadsUseCase(
    private val gateway: MessageThreadGateway,
) {
    suspend operator fun invoke(): List<MessageThreadSummary> = gateway.listThreads()
}

File: client/ios/UseCase/Message/Thread/ListMessageThreadsUseCase.swift
import Foundation

// Marketing America Corp. Oleksandr Tishchenko
public struct ListMessageThreadsUseCase: Sendable {
    private let gateway: MessageThreadGateway

    public init(gateway: MessageThreadGateway) {
        self.gateway = gateway
    }

    public func callAsFunction() async throws -> [MessageThreadSummary] {
        try await gateway.listThreads()
    }
}
